src/coreclr/src/jit/gentree.cpp

```diff
@@ -5928,11 +5928,19 @@ GenTree* Compiler::gtNewStringLiteralNode(InfoAccessType iat, void* pValue)
         case IAT_PVALUE: // The value needs to be accessed via an indirection
             // Create an indirection
             tree = gtNewIndOfIconHandleNode(TYP_REF, (size_t)pValue, GTF_ICON_STR_HDL, false);
+#ifdef DEBUG
+            tree->gtGetOp1()->AsIntCon()->gtMethodHandle =
+                (CORINFO_METHOD_HANDLE)GenTreeIntCon::MethodHandleType::StringLiteralNode;
```

Comment:
This is a case where hopefully (in a follow-up) we could store the actual string handle and then display the actual string (or at least a prefix, for long strings) in the asm output.

Don't you need another case for the `IAT_VALUE` case above?

Reply:
Yes, fixed it although, today `eeGetCPString()` method used to get string literal from `stringHandle` is not supported on linux/arm. I have added a note.

Reply:
Also, I didn't notice the location of `IAT_VALUE` being used under `adrp` so didn't add. But I think it makes sense to add it there.